kernel/include/kernel_structs.h

```diff
@@ -118,6 +118,11 @@ struct _kernel {
 	struct k_thread *threads; /* singly linked list of ALL fiber+tasks */
 #endif
 
+#if defined(CONFIG_USERSPACE)
+	/* 0 bits for ids currently in use, 1 for free ids */
+	u8_t free_thread_ids[CONFIG_MAX_THREAD_BYTES];
```

Comment:
wouldn't it be easier if we use bool type here? like
bool free_thread_ids[CONFIG_MAX_THREADS];
The efforts to set and clear the thead_id will be less if we use bool type.

Reply:
While this makes the access more efficient, it also wastes memory.  There's just one _kernel struct, so it's not a lot of memory, but bitmaps are already used elsewhere almost everywhere else where we need a boolean array.

Reply:
Ok got it. thanks.